### PR TITLE
Misfire instructions support

### PR DESCRIPTION
**Pull request by Gavrilov-Ivan**
See #2160 
Update:
Do not merge due to https://github.com/jmix-framework/jmix/issues/2160#issuecomment-1723077114

### CODE REVIEW

File: jmix-quartz/quartz-flowui/src/main/resources/io/jmix/quartzflowui/messages.properties [hunk 1]
```diff
@@ -68,5 +68,30 @@ io.jmix.quartzflowui.view.trigger/triggerRepeatModeHelpText=<div>Repeat mode def
                 </ul>\
                 'Finite Repeats' and 'Execute forever' modes are limited by the End Date (if exists).\
             </div>
+io.jmix.quartzflowui.view.trigger/triggerMisfireInstructionHelpText=<div>Misfire instruction defines behavior if trigger misfire is detected.\
```
[the remark this answers is not in the document]
Comment (Gavrilov-Ivan) [reply jmix-quartz/quartz-flowui/src/main/resources/io/jmix/quartzflowui/messages.properties]: Fixed 'description' text, keep current options names (it doesn't look like it's required an extra articles there. And Chat GPT 'fixed' the same phrase the different ways).

File: jmix-quartz/quartz-flowui/src/main/resources/io/jmix/quartzflowui/view/trigger/trigger-model-detail-view.xml [hunk 2]
```diff
@@ -32,6 +32,12 @@
             <textField id="repeatIntervalField" property="repeatInterval" required="true" visible="false"/>
             <dateTimePicker id="startDateTimePicker" property="startDate"/>
             <dateTimePicker id="endDateTimePicker" property="endDate"/>
+            <hbox id="misfireInstructionBox">
```
[the remark this answers is not in the document]
Comment (Gavrilov-Ivan) [reply jmix-quartz/quartz-flowui/src/main/resources/io/jmix/quartzflowui/view/trigger/trigger-model-detail-view.xml]: Issues due to new rules of the FormLayout. Fixed.

File: jmix-quartz/quartz/src/main/java/io/jmix/quartz/model/CronTriggerMisfireInstruction.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+import io.jmix.core.metamodel.datatype.EnumClass;
+import org.quartz.CronScheduleBuilder;
+import org.quartz.CronTrigger;
+import org.springframework.lang.Nullable;
+
+import java.util.function.Consumer;
+
+
+public enum CronTriggerMisfireInstruction implements EnumClass<String> {
```
[the remark this answers is not in the document]
Comment (Gavrilov-Ivan) [reply jmix-quartz/quartz/src/main/java/io/jmix/quartz/model/CronTriggerMisfireInstruction.java:31]: Done

File: jmix-quartz/quartz/src/main/java/io/jmix/quartz/model/CronTriggerMisfireInstruction.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    public String getId() {
+        return id;
+    }
+
+    public int getCode() {
+        return code;
+    }
+
+    public void applyInstruction(CronScheduleBuilder builder) {
```
[the remark this answers is not in the document]
Comment (Gavrilov-Ivan) [reply jmix-quartz/quartz/src/main/java/io/jmix/quartz/model/CronTriggerMisfireInstruction.java:71]: Done

File: jmix-quartz/quartz/src/main/java/io/jmix/quartz/model/CronTriggerMisfireInstruction.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            CronTrigger.MISFIRE_INSTRUCTION_FIRE_ONCE_NOW),
+    DO_NOTHING("do_nothing",
+            CronScheduleBuilder::withMisfireHandlingInstructionDoNothing,
+            CronTrigger.MISFIRE_INSTRUCTION_DO_NOTHING);
+
+    private final String id;
+    private final Consumer<CronScheduleBuilder> consumer;
+    private final int code;
+
+    CronTriggerMisfireInstruction(String id, Consumer<CronScheduleBuilder> consumer, int code) {
```
[the remark this answers is not in the document]
Comment (Gavrilov-Ivan) [reply jmix-quartz/quartz/src/main/java/io/jmix/quartz/model/CronTriggerMisfireInstruction.java:53]: Done

File: jmix-quartz/quartz/src/main/java/io/jmix/quartz/model/SimpleTriggerMisfireInstruction.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import io.jmix.core.metamodel.datatype.EnumClass;
+
+import org.quartz.SimpleScheduleBuilder;
+import org.quartz.SimpleTrigger;
+import org.springframework.lang.Nullable;
+
+import java.util.function.Consumer;
+
+
+public enum SimpleTriggerMisfireInstruction implements EnumClass<String> {
```
[the remark this answers is not in the document]
Comment (Gavrilov-Ivan) [reply jmix-quartz/quartz/src/main/java/io/jmix/quartz/model/SimpleTriggerMisfireInstruction.java:31]: Done

File: jmix-quartz/quartz/src/main/java/io/jmix/quartz/service/QuartzService.java [hunk 7]
```diff
@@ -233,10 +253,34 @@ private Trigger buildTrigger(JobDetail jobDetail, TriggerModel triggerModel) {
...
 
         if (triggerModel.getScheduleType() == ScheduleType.CRON_EXPRESSION) {
-            triggerBuilder.withSchedule(cronSchedule(triggerModel.getCronExpression()));
+            String cronExpression = triggerModel.getCronExpression();
+            if(cronExpression == null) {
+                throw new IllegalStateException("Cron trigger has null cron expression");
+            }
+            CronScheduleBuilder cronScheduleBuilder = cronSchedule(cronExpression);
+            String misfireInstructionId = triggerModel.getMisfireInstructionId();
+            if(misfireInstructionId != null) {
```
[the remark this answers is not in the document]
Comment (Gavrilov-Ivan) [reply jmix-quartz/quartz/src/main/java/io/jmix/quartz/service/QuartzService.java]: Done

**Review by gorbunkov — APPROVED** [review]

**Pull request merged**